lib/core/base/virtual-node/serial-virtual-node.js

```diff
@@ -44,14 +44,48 @@ class SerialVirtualNode extends AbstractVirtualNode {
   }
 }
 
+/**
+ * Map nodeName to nodeType
+ */
+const NodeNamesToTypesEnum = Object.freeze({
+  '#cdata-section': 2,
+  '#text': 3,
+  '#comment': 8,
+  '#document': 9,
+  '#document-fragment': 11
+});
+
 /**
  * Convert between serialised props and DOM-like properties
  * @param {Object} serialNode
  * @return {Object} normalProperties
  */
 function normaliseProps(serialNode) {
   let { nodeName } = serialNode;
-  const { nodeType = 1 } = serialNode;
+  if (serialNode.nodeType === 2) {
```

Comment:
This could be simplified to look at the object and return the value. If you reverse the values and keys of the object you can also do the same for nodeName.

```js

```js
const nodeTypeToName = {};
const keys = Object.keys(nodeNamesToTypes);
Object.values(nodeNamesToTypes).forEach((value, index) => {
  nodeTypeToName[value] = keys[index];
});

const nodeType = serialNode.nodeType ?? nodeNamesToTypes[serialNode.nodeName] ?? 1;
const nodeName = serialNode.nodeName ?? nodeTypeToName[serialNode.nodeType];
```